docs/excel/custom-functions-best-practices.md

```diff
@@ -0,0 +1,184 @@
+---
+ms.date: 09/18/2018
+description: Learn best practices and recommended patterns for Excel custom functions.
+title: Custom Functions' best practices
+---
+
+# Custom Functions' best practices
+
+This article covers some recommended patterns and solutions to common use cases with Excel custom functions.
+
+## Authentication
+
+Non-custom functions add-ins which have web views (i.e. display information in the taskpane) accomplish authentication via `Office.context.displayDialogAsync`. Custom functions differ from this approach. Because custom functions use a different runtime, they do not have access to the Office.context methods. Instead, they use methods available on the `OfficeRuntime` object, such as `OfficeRuntime.DisplayWebDialog()` and `OfficeRuntime.AsyncStorage.setItem(key, value)`.
+
+The following code sample shows how to store a token in AsyncStorage and display a dialog box which can indicate whether or not the user is authenticated.  
```

Comment:
This description is not accurate "The following code sample shows how to store a token in AsyncStorage and display a dialog box which can indicate whether or not the user is authenticated."

For one thing, it does NOT show "how to store". Instead, the getToken method just _assumes_ that _cachedToken is a reference to some stored token. Which is mysterious, because the `else` clause of the getToken method _does not store the token_ anywhere. This looks suspiciously like untested code. 
Secondly, the dialog box does not "indicate whether or not the user is authenticated". Instead it opens the login page at the mythical www.auth.com. Unless the user is already authenticated (that is, there is already a token) in which case the dialog does not open at all.

Reply:
Yeah, you're right. I apologize I didn't do a better job of catching this glaring error the first time around. I went back to engineering today and got another sample that I think makes more sense.